tests/system_tests_three_routers.py

```diff
@@ -220,6 +225,71 @@ def run(self):
         Container(self).run()
  
  
+class DynamicRequestResponseTest(MessagingHandler):
+    def __init__(self, address1, address2):
+        super(DynamicRequestResponseTest, self).__init__(prefetch=0)
+        self.address1   = address1
+        self.address2   = address2
+        self.dest       = "closest.whatever"
+        self.error      = None
+        self.sender     = None
+        self.server_receiver = None
+        self.client_receiver = None
+        self.n_expected = 10
+        self.n_sent     = 0
+        self.n_received = 0
+        self.n_accepted = 0
+ 
+    def timeout(self):
+        self.error = "Timeout Expired %d messages received." % self.n_received
+        self.conn1.close()
+        self.conn2.close()
+ 
+    def on_released(self, event):
+        self.n_sent -= 1
+        time.sleep(0.1)
+
+    def on_link_opened(self, event):
+        if event.receiver:
+            self.sender = event.container.create_sender(self.conn1, None)
+
+    def on_start(self, event):
+        self.timer = event.reactor.schedule(5, Timeout(self))
+        self.conn1 = event.container.connect(self.address1)
+        self.conn2 = event.container.connect(self.address2)
+        self.server_receiver = event.container.create_receiver(self.conn2, self.dest)
+        self.client_receiver = event.container.create_receiver(self.conn2, None, dynamic=True)
```

Comment:
Both the client and server receivers are established on the same connection.  Is this what was intended?  I would think that one connection would be the client and the other the server.